### PR TITLE
fix dims mixing

### LINKED ISSUE #432: mixing dims
## 🐛 Bug

Adding a 4D image followed by a 5D image causes the sliders to get really messed up, need to fix inside `qt_dims.py`

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR fixes #432 and other bugs we've been having with our dims display when the range of a slider is set to 1, needed for starFISH amongst other tools @kne42 

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [x] add more example tests including `examples/4D_image_5D_image.py`

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: napari/components/viewer_model.py [hunk 1]
```diff
@@ -446,9 +446,9 @@ def add_volume(self, volume, *args, **kwargs):
         layer = layers.Volume(volume, *args, **kwargs)
         if self.dims.ndim == 2:
             self.dims.ndim = 3
-        self.dims.set_display(-3, True)
+        self.dims.set_display(self.dims.ndim - 3, True)
```
Comment (kne42) [inline napari/components/viewer_model.py]: shouldn't this be changed inside `self.dims.set_display` to perform this "wrap" when provided a negative number instead?

**Review by kne42 — APPROVED** [review]

File: napari/components/viewer_model.py (hunk 1 above)
Comment (sofroniewn) [reply napari/components/viewer_model.py]: fixed

**Comment by sofroniewn** [comment]
@kne42 have you been able to fix starfish problems with this?

**Comment by kne42** [comment]
i have confirmed that it works in https://github.com/spacetx/starfish/pull/1430

**Pull request merged**